helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java

```diff
@@ -460,7 +438,33 @@ public void onError(Exception e, ErrorCode code, ErrorType type) {
       StateTransitionError error = new StateTransitionError(type, code, e);
       _stateModel.rollbackOnError(_message, _notificationContext, error);
     }
+  }
+
+  // Verify the fromState and current state of the stateModel.
+  public Exception staleMessageValidator() {
```

Comment:
`staleMessageValidator` sounds more like a class/variable name. How about changing the method name to a verb style: `validateStaleMessage()` (it does something)?

Reply:
In my understanding, validateXXX is a function throws exception and returns void or boolean. Since I have a function that returns an exception, I am not sure about the name validateXXX.

Reply:
nit, it would be more graceful to define a ValidationResult class and contain a boolean (pass or not) and an optional Exception. Especially as a public method. Is there a chance we can make it protected or package private?

Reply:
We are not using the boolean return value. May I ask is this for a cleaner interface purpose for further usage?

Reply:
Updated.